File: spring-boot-java/src/main/java/io/ssaragi/java/spring/boot/exception/BaseException.java
package io.ssaragi.java.spring.boot.exception;

import lombok.Getter;

@Getter
public abstract class BaseException extends RuntimeException {

    // 결과 코드 및 메세지
    private final String code;
    private final String message;

    // 트랜잭션 ID
    private final String trId;

    public BaseException(String trId, String code, String message) {
        this.code = code;
        this.message = message;
        this.trId = trId;
    }
}

File: spring-boot-kotlin/src/main/kotlin/io/ssaragi/kotlin/spring/boot/exception/BaseException.kt
package io.ssaragi.kotlin.spring.boot.exception

open class BaseException(

    // 결과 코드 및 메세지
    val code: String,
    override val message: String,

    // 트랜잭션 ID
    val trId: String,

) : RuntimeException(message)
